src/mydumper_start_dump.c

```diff
@@ -1248,26 +1263,11 @@ void start_dump() {
   fclose(mdfile);
```

Comment:
Here is where mdfile is being close, but you are streaming metadata before closing. 
The stream backup works like this:
- you open the file and save all the info
- you close the file to guarantee that everything is in there
- you send the file to the stream thread, which is going to reopen, send the file through the stdout and close it again
You should move the close before enqueueing into the stream queue if nothing is been writing in mdfile

Reply:
> you close the file to guarantee that everything is in there

Then I guess flushing must be enough. There are chicken and egg problem for doing this after close.